osu.Game/Screens/BackgroundMode.cs

```diff
@@ -81,10 +80,25 @@ protected override void OnSuspending(GameMode next)
 
         protected override bool OnExiting(GameMode next)
         {
+            if (base.OnExiting(next)) return true;
+
             Content.FadeOut(transition_length, EasingTypes.OutExpo);
             Content.MoveToX(x_movement_amount, transition_length, EasingTypes.OutExpo);
 
-            return base.OnExiting(next);
+            stopPendingLoad();
+            return false;
+        }
+
+        public override void MakeCurrent()
+        {
+            base.MakeCurrent();
+            stopPendingLoad();
+        }
+
+        private void stopPendingLoad()
+        {
+            pendingLoad?.Exit();

```

Comment:
I don't like `Exit` being able to run concurrently with `load`. I think _absolutely nothing_ related to the loading drawable should run concurrently with `load` or we will quickly get into the mess of having to make tons of things threadsafe. I would much rather keep the load abortion logic contained within `load` by adding another volatile boolean just for that specific purpose.